src/widget/woverview.cpp

```diff
@@ -737,8 +739,10 @@ void WOverview::drawWaveformPixmap(QPainter* pPainter) {
         if (normalize && m_pixmapDone && m_waveformPeak > 1) {
             diffGain = 255 - m_waveformPeak - 1;
         } else {
+            DEBUG_ASSERT(m_pCurrentTrack);
```

Comment:
How about using `VERIFY_OR_DEBUG_ASSERT` so you can prevent the segfault at line 745?

Reply:
`m_waveformSourceImage` ic cleared when a track is loaded (null or track) so we shouldn't get here with null track in the first place. So I think we're good with a DEBUG_ASSERT like in the other places.

Reply:
In this case, how about removing this `DEBUG_ASSERT`? Any reasons why you added it in a first place?

Reply:
My understanding is that DEBUG_ASSERT should be used where we are sure a certain precondition is met at the time of writing, but surrounding code / callers may be changed later on and programming errors should be catched in debug builds.

Otoh VERIFY_OR_DEBUG_ASSERT is for when we are not 100% sure.

This targets main so there's plenty of time to discover situations where the assumption is wrong.
Does that make sense?

The only dev discussion about this that I found is https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/Bpm.3A.3Avalue.28.29.20without.20checking.20validity

Reply:
It does make sense, thanks for clarifying!